src/lib/y2security/security_policies/policy.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2security/security_policies/validator"
+
+module Y2Security
+  module SecurityPolicies
+    # This class represents a security policy
+    #
+    # It offers an API to get the security policies and run validations.
+    #
+    # @example Get all known security policies
+    #   Policy.all #=> [#<Y2Security::Policy...>]
+    #   Policy.all.map(&:name) #=> ["Defense Information Systems Agency STIG"]
+    #
+    # @example Run DISA STIG networking validation
+    #   policy = Policy.find(:disa_stig)
+    #   policy.validate.map(&:to_message) #=> ["Wireless devices are not allowed"]
+    class Policy
+      # @return [Symbol] Security policy ID
+      attr_reader :id
+      # @return [String] Security policy name
+      attr_reader :name
+      # @return [Array<String>] Security policy packages needed
+      attr_reader :packages
+      # @return [Array<Issue>] Found security policy issues
+      attr_reader :issues
+
+      class << self
+        # Returns the list of known security policies
+        #
+        # @return [Array<Policy>]
+        def all
```

Comment:
This class stores a list of policy objects and it also allows to create other new polices. But the new policies are not contained in the list of policies:

~~~
my_policy = Policy.new(:my_policy, "My security policy")
Policy.all.map(&:id) #=> [:disa_stig]
~~~

I would set the constructor as private method.

Reply:
Regarding this issue, I do not like the idea of making the constructor private. Perhaps we could find a better name for the `.all` method (something like `.known`). Or we could provide an API to register policies so they are listed as part of that class method.